cpp/dolfinx/mesh/Topology.h

```diff
@@ -89,11 +115,40 @@ class Topology
   std::shared_ptr<const graph::AdjacencyList<std::int32_t>>
```

Comment:
Probably needs a note for now that this should only be used for single cell type meshes?

Reply:
I've added a check to throw an error if misused. Trying not to touch the existing interface otherwise.